Content.Server/GameObjects/Components/Atmos/FirelockComponent.cs

```diff
@@ -89,7 +90,7 @@ public override async Task<bool> InteractUsing(InteractUsingEventArgs eventArgs)
 
                 if (State == DoorState.Closed && !IsWeldedShut)
                     Open();
-                else if (State == DoorState.Open)
+                else if (State == DoorState.Open && !IsWeldedShut)
```

Comment:
Instead of checking this twice... Encapsulare this in a `if (!IsWeldedShut)` check.